src/frontend/src/components/render/Instance.tsx

```diff
@@ -30,39 +31,39 @@ export function RenderInstance({
   model,
   instance
 }: {
-  model: string;
+  model: ModelType | undefined;
   instance: any;
 }): ReactNode {
   switch (model) {
-    case 'address':
+    case ModelType.address:
```

Comment:
Instead of this big switch case skyscraper you could define an object why acts like a map. Then the arguments of the rendered would all need to be named the same, something like data, and then just lookup if the type is in the object and then render the value as component.